test/integration/integration_test.go

```diff
@@ -8,9 +8,43 @@ import (
 	"github.com/kyma-project/helm-broker/pkg/apis/addons/v1alpha1"
 )
 
+func TestHttpBasicAuth(t *testing.T) {
+	// todo: uncomment after templating is done
+	t.Skip()
+	// given
+	suite := newTestSuite(t, HTTPBasicAuthEnabled)
```

Comment:
I like that u used the functional opts for creators like `createAddonsConfiguration`


but IMO using that in `newTestSuite` can make test unreadable
having sth like that:
```go
	suite := newTestSuite(t, config{
		docsEnabled:   true,
		httpBasicAuth: false,
	})
```

allows me to see full test setup and I know what I can expect. With `suite := newTestSuite(t, HTTPBasicAuthEnabled)` and do not know what to expect, docs enabled or not? without knowing the default it's not clear, so it forces me to go to impl details of the `newTestSuite`

and for two params having the normal func arg is also good enough
```go
 	suite := newTestSuite(t, /*docsEnabled*/ true, /*httpBasicAuth*/ false)
```